internal/handlers/common/required.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// GetRequiredParam returns doc's value for key or protocol error for missing or invalid parameter.
+func GetRequiredParam[T types.ScalarType](doc *types.Document, key string) (T, error) {
```

Comment:
It seems that `GetRequiredParam` tightly bounded with `types.Document`. Maybe we should place it somewhere near `types.Document` declaration?

Reply:
It returns handler-level error. Essentially, that's request validator

Reply:
Ok then.